src/models/BanList.ts

```diff
@@ -139,7 +139,9 @@ class BanList extends EventEmitter {
         if (stateKeyMap) {
             for (const event of stateKeyMap.values()) {
                 const rule = event?.unsigned?.rule;
-                if (rule && rule.kind === kind) {
+                // README! If you are refactoring this, you certainly will want
```

Comment:
I don't understand this comment.

Reply:
The point is, in the future we probably will want a way of getting the rules from a `BanList` that have a different `recommendation` to `m.ban`, however I really want to avoid a situation where someone decides to refactor this method to serve that purpose, even though currently it is only used in contexts where the rule must have the recommendation `m.ban` or we will cause a terrible mistake.

Reply:
FTR: it was already returning every rule regardless of recommendation, despite all the callers wanting only `m.ban`.

Reply:
So, would the following be consistent with what you have in mind?

```
// README! If you are refactoring this and/or introducing a mechanism to return the list of rules, please make sure that you *only* return rules with `m.ban`!
```